internal/gatewayapi/listener.go

```diff
@@ -104,6 +104,10 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR XdsIRMap
 					Address: "0.0.0.0",
 					Port:    uint32(containerPort),
 					TLS:     irTLSConfigs(listener.tlsSecrets),
+					Path: ir.PathSettings{
```

Comment:
maintaining logic in two places in the gatewayapi layer may be harder to reason with for the next dev trying to figure out translation, thoughts ?
instead if `ir.path` is `nil`, the xds translator layer could set defaults

Reply:
> maintaining logic in two places in the gatewayapi layer may be harder to reason with for the next dev trying to figure out translation, thoughts ?

I think the IR should encapsulate what is expected of the XDS layer, and all defaults should be set in the IR layer. There should be no magic translations happening in the XDS layer - XDS translation should be a technical translation without filling in any controlled attribute.

> instead if ir.path is nil, the xds translator layer could set defaults

By that logic, should I also remove the `Address` default of `0.0.0.0` in this structure (multiple places in the `gatewayapi` folder) and the default path being set in [processAccessLog](https://github.com/envoyproxy/gateway/blob/main/internal/gatewayapi/listener.go#L168)?